src/model/cloud-runner/aws/aws-job-stack.ts

```diff
@@ -0,0 +1,186 @@
+        ParameterValue: 'echo "this template should be overwritten when running a task"',
+      },
+      {
+        ParameterKey: 'EntryPoint',
+        ParameterValue: entrypoint.join(','),
+      },
+      {
+        ParameterKey: 'WorkingDirectory',
+        ParameterValue: workingdir,
+      },
+      {
+        ParameterKey: 'EFSMountDirectory',
+        ParameterValue: mountdir,
+      },
+      {
+        ParameterKey: 'BUILDID',
+        ParameterValue: buildGuid,
+      },
+      ...secretsMappedToCloudFormationParameters,
+    ];
+
+    let previousStackExists = true;
+    while (previousStackExists) {
+      previousStackExists = false;
+      const stacks = await CF.listStacks().promise();
+      if (!stacks.StackSummaries) {
+        throw new Error('Faild to get stacks');
+      }
+      for (let index = 0; index < stacks.StackSummaries.length; index++) {
+        const element = stacks.StackSummaries[index];
+        if (element.StackName === taskDefStackName && element.StackStatus !== 'DELETE_COMPLETE') {
+          previousStackExists = true;
+          CloudRunnerLogger.log(`Previous stack still exists: ${JSON.stringify(element)}`);
+        }
+      }
+    }
+
+    try {
+      await CF.createStack({
+        StackName: taskDefStackName,
+        TemplateBody: taskDefCloudFormation,
+        Capabilities: ['CAPABILITY_IAM'],
+        Parameters: parameters,
+      }).promise();
+      CloudRunnerLogger.log('Creating cloud runner job');
+      await CF.createStack({
+        StackName: cleanupTaskDefStackName,
+        TemplateBody: cleanupCloudFormation,
+        Capabilities: ['CAPABILITY_IAM'],
+        Parameters: [
+          {
+            ParameterKey: 'StackName',
+            ParameterValue: taskDefStackName,
+          },
+          {
+            ParameterKey: 'DeleteStackName',
+            ParameterValue: cleanupTaskDefStackName,
+          },
+          {
+            ParameterKey: 'TTL',
+            ParameterValue: '100',
+          },
+          {
+            ParameterKey: 'BUILDID',
+            ParameterValue: buildGuid,
+          },
+          {
+            ParameterKey: 'EnvironmentName',
+            ParameterValue: this.baseStackName,
+          },
+        ],
+      }).promise();
+      await CF.waitFor('stackCreateComplete', { StackName: taskDefStackName }).promise();
+    } catch (error) {
+      await AWSError.handleStackCreationFailure(
+        error,
+        CF,
+        taskDefStackName,
+        //taskDefCloudFormation,
+        //parameters,
+        //secrets,
+      );
+      throw error;
+    }
+
+    const taskDefResources = (
+      await CF.describeStackResources({
+        StackName: taskDefStackName,
+      }).promise()
+    ).StackResources;
+
+    const baseResources = (await CF.describeStackResources({ StackName: this.baseStackName }).promise()).StackResources;
+
+    // TODO: offer a parameter to decide if you want the guaranteed shutdown or fastest startup time possible
```

Comment:
How much difference is there between these two options?

Depending on the difference reliability might be desired in CI.

Reply:
Great question, will quantify the difference specifically.

Reply:
Would like to leave this TODO comment and include this implementation in the next iteration @webbertakken